### PR TITLE
remove IFakeCreatorFacade and DefaultFakeCreatorFacade

### PR DESCRIPTION
**Pull request by adamralph**

### CODE REVIEW

File: src/FakeItEasy/Sdk/Create.cs [hunk 1]
```diff
@@ -36,7 +36,10 @@ public static object Fake(Type typeOfFake)
         /// <returns>A fake object.</returns>
         public static object Fake(Type typeOfFake, Action<IFakeOptions> optionsBuilder)
         {
-            return FakeCreator.CreateFake(typeOfFake, optionsBuilder);
+            Guard.AgainstNull(typeOfFake, nameof(typeOfFake));
+            Guard.AgainstNull(optionsBuilder, nameof(optionsBuilder));
```
Comment (adamralph) [inline src/FakeItEasy/Sdk/Create.cs]: I don't think these guards belong here but, in folding `DefaultFakeCreatorFacade` into this type, I've brought them along for the ride. I'll address my concerns here in a later PR.

**Comment by adamralph** [comment]
Some may be alarmed at the removal of tests in this PR, but I believe the ones I have removed are classic examples of [Gulliver's Travels Tests](http://dhickey.ie/2014/03/gullivers-travels-test/) and add no value.

**Comment by blairconrad** [comment]
> Some may be alarmed at the removal of tests in this PR,

Nah, I'm pretty sanguine about that sort of thing, especially if I think that the Specs are covering the cases.

Thanks for the introduction to the colourful test name.

File: src/FakeItEasy/Sdk/Create.cs [hunk 2]
```diff
@@ -72,7 +75,9 @@ public static IList<object> CollectionOfFake(Type typeOfFake, int numberOfFakes,
         [EditorBrowsable(EditorBrowsableState.Advanced)]
         public static object Dummy(Type typeOfDummy)
         {
-            return FakeCreator.CreateDummy(typeOfDummy);
+            Guard.AgainstNull(typeOfDummy, nameof(typeOfDummy));
+
+            return FakeAndDummyManager.CreateDummy(typeOfDummy);
```
Comment (blairconrad) [inline src/FakeItEasy/Sdk/Create.cs]: Same "don't belong here" comment?

**Comment by blairconrad** [comment]
@adamralph, it all looks good to me. I'll give @thomaslevesque a bit of time, just in case he has a concern, but I'm happy to merge.

File: src/FakeItEasy/Sdk/Create.cs (hunk 2 above)
Comment (adamralph) [inline src/FakeItEasy/Sdk/Create.cs]: Yep.

**Comment by thomaslevesque** [comment]
LGTM, thanks @adamralph!

**Pull request merged**